### ISSUE TITLE
URI constructor does not take an object as value for query property

### ISSUE THREAD

**Issue by wachunei** [medialize/URI.js#366]
Hello everyone,

Please let me know if this is this has been decided to be like this or not, but when building an URL, query property only takes a string as value, instead of search() or query() that take both string and objects. 

Heres an example code ([online repl here](https://repl.it/repls/AshamedRoundPangolin)):
```js 
const assert = require('assert');
const URI = require('urijs');

const PROTOCOL = 'http';
const HOST = 'mycustom.domain.com';
const PORT = '443';
const BASE_PATH = '/some//path/';
const resource = '///resource//5';
const query = {
  myQuery: 'rocks'
}

const constructorURI = new URI({
    protocol: PROTOCOL,
    hostname: HOST,
    port: PORT,
    path: `${BASE_PATH}/${resource}/`,
    query,
  }).normalize();
  
  
const searchURI = new URI({
    protocol: PROTOCOL,
    hostname: HOST,
    port: PORT,
    path: `${BASE_PATH}/${resource}/`,
  }).search(query).normalize();

assert(constructorURI.toString() === searchURI.toString(), 'URIs are not equal');
```

**Comment by rodneyrehm** [comment]
Hey there!

Providing an object was originally meant to be used internally for convenient cloning. But since it's [public API](http://medialize.github.io/URI.js/docs.html#constructor) I guess we may want to make this a little more consistent.

You could extend [`.href()`](https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L1241-L1248) to not assign `query` to `._parts` directly. After all the remaining properties have been processed you could call `.query(value, build)` and thus allow passing in an object. It's important to call `.query()` after `._parts` has been populated in order for the properties `duplicateQueryParameters` and `escapeQuerySpace` to have been imported properly.

With this change you'd also be able to pass `URI({ query: '?some=data' })`, which, according to the docs, shouldn't be supported right now either.

I guess this change should not only apply to `query`, but also to `fragment` in order to properly support the [fragment abuse plugins](http://medialize.github.io/URI.js/docs.html#fragment-abuse).

Would you like to submit a PR for this change?

**Comment by wachunei** [comment]
I'm glad to actually find an opportunity to contribute 😄 

Will look into this in the next few days, In case of any doubt I'll comment here, if you don't mind.

**Comment by wachunei** [comment]
I was a bit lost, I don't know if what I made makes total sense, let me know if I need to fix something.

**Comment by wachunei** [comment]
closed by #367

**Comment by rodneyrehm** [comment]
released as [v1.19.1](https://github.com/medialize/URI.js/releases/tag/v1.19.1)
